### PR TITLE
executors: Update Terraform Deploy Docs

### LINKED ISSUE #43011: executors: Fix up Terraform Deployment Docs
While [setting up executors](https://github.com/sourcegraph/infrastructure/pull/4078) for the scaletesting instance, I dogfood our [terraform deployment docs](https://docs.sourcegraph.com/admin/deploy_executors_terraform). I ran into a few confusions while setting everything up (thus slowing me down and making me unsure about what I was doing).

## Updates

* Add a table or list of the possible `<cloud>` values. 
    * Be nice to associate values with links to the terraform module repos
* Update references to all variables to match what the user would actually use (for example, instead `min_replicas` it should be `executor_min_replicas`)
* Revisit the step-by-step installation guide to ensure everything still makes sense
    * If a user is setting up a batches executor, they probably do not need to set `codeIntelAutoIndexing.enabled` in the site config
* Update references to `4.0.0` to `4.0.1` (there is a bug in `4.0.0`)
* Autoscaling should have `metric_writer_credentials_file` with `sensitive = true`
    * User needs to run `terraform output  metric_writer_credentials_file` to get value
    * This may apply to the AWS values as well
* Specify what roles a user must have to set everything up in GCP and AWS 
    * For example in GCP, user must have `compute.disks.create`, `iam.serviceAccounts.create`, `compute.networks.create`, and permissions to create a role in the project (scaling requires Secret Manager permissions). There are probably more roles a user needs beyond these

### PR DESCRIPTION
**Pull request by Piszmog**
Closes #43011.

Based on my experience standing up executors in the scaletesting instance, I updated to Terraform Deploy docs to match my experience and to clarify unknowns I had as I wan thru the documentation.

## Test plan

Ran the doc site locally to view changes.

### CODE REVIEW

File: doc/admin/deploy_executors_terraform.md [hunk 1]
```diff
@@ -8,216 +8,417 @@
...
+| `executor_metrics_environment_label`           | The value for environment by which to filter the custom metrics.                                                                                                                                                   |
+
+See the Terraform Modules for additional configurations.
+
+- [AWS](https://sourcegraph.com/github.com/sourcegraph/terraform-aws-executors/-/blob/modules/executors/variables.tf)
+- [Google](https://sourcegraph.com/github.com/sourcegraph/terraform-google-executors/-/blob/modules/executors/variables.tf)
+
+## Terraform Version
+
+The executor Terraform modules require Terraform 1.1. Use [tfenv](https://github.com/tfutils/tfenv) to install Terraform
```
Comment (eseliger) [inline doc/admin/deploy_executors_terraform.md:60]: Maybe I misunderstood how to read those constraints but I thought it would mean 1.1 and any above lol

Comment (eseliger) [reply doc/admin/deploy_executors_terraform.md:60]: If not, I wonder if we shouldn't do that and just say here, use 1.1+

File: doc/admin/deploy_executors_terraform.md [hunk 2]
```diff
@@ -8,216 +8,417 @@
...
+- secretmanager.locations.list
+- secretmanager.secrets.*
+- secretmanager.versions.*
+
+## Isolate Commands
+
+We perform layered security/security in-depth at untrusted boundaries. Untrusted code is run only within a fresh virtual
+machine, and the host machine running untrusted code does not have privileged access to the Sourcegraph instance.
+
+To isolate commands in virtual machines, set `executor_use_firecracker` to true.
```
Comment (eseliger) [inline doc/admin/deploy_executors_terraform.md:130]: I think this is the default, isn't it?

**Review by eseliger — APPROVED** [review]
good stuff!

File: doc/admin/deploy_executors_terraform.md (hunk 2 above)
Comment (Piszmog) [reply doc/admin/deploy_executors_terraform.md:130]: Yep it is. I'll remove this section (thought it was not defaulting to `true`)

**Pull request merged**